src/main/java/org/killbill/billing/plugin/vertex/dao/VertexResponseDataExtractor.java

```diff
@@ -0,0 +1,323 @@
+
+    List<AddressInfo> getAddresses() {
+        return addresses;
+    }
+
+    BigDecimal getTotalTaxCalculated() {
+        return totalTaxCalculated;
+    }
+
+    BigDecimal getTotalTaxExempt() {
+        return totalTaxExempt;
+    }
+
+    BigDecimal getTotalTaxable() {
+        return totalTaxable;
+    }
+
+    String getDocumentCode() {
+        return vertexResponseData.getTransactionId();
+    }
+
+    LocalDateTime getDocumentDate() {
+        return vertexResponseData.getDocumentDate() != null ? vertexResponseData.getDocumentDate().atStartOfDay() : null;
+    }
+
+    BigDecimal getTotalAmount() {
+        return vertexResponseData.getTotal() != null ? BigDecimal.valueOf(vertexResponseData.getTotal()) : null;
+    }
+
+    BigDecimal getTotalDiscount() {
+        return vertexResponseData.getDiscount() != null && vertexResponseData.getDiscount().getDiscountValue() != null
+               ? BigDecimal.valueOf(vertexResponseData.getDiscount().getDiscountValue())
+               : null;
+    }
+
+    BigDecimal getTotalTax() {
+        return vertexResponseData.getTotalTax() != null ? BigDecimal.valueOf(vertexResponseData.getTotalTax()) : null;
+    }
+
+    LocalDateTime getTaxDate() {
+        return vertexResponseData.getTaxPointDate() != null ? vertexResponseData.getTaxPointDate().atStartOfDay() : null;
+    }
+
+    List<OwnerResponseLineItemType> getTaxLines() {
+        return vertexResponseData.getLineItems();
+    }
+
+    private List<TaxInfo> buildTransactionSummary() {
+        final List<OwnerResponseLineItemType> lineItems = vertexResponseData.getLineItems();
+        return lineItems != null
+               ? lineItems.stream()
+                          .filter(Objects::nonNull)
+                          .map(OwnerResponseLineItemType::getTaxes)
+                          .filter(Objects::nonNull)
+                          .flatMap(Collection::stream)
+                          .map(TaxInfo::new)
+                          .collect(Collectors.toList())
+               : ImmutableList.of();
+    }
+
+    private Map<String, Object> buildAdditionalData() {
+        final Map<String, Object> additionalData = new HashMap<>();
+
+        additionalData.put("roundAtLineLevel", vertexResponseData.getRoundAtLineLevel()); //todo: add additional useful information if needed
+
+        return additionalData;
+    }
+
+    private List<AddressInfo> buildAddresses() {
+        final CustomerType customer = vertexResponseData.getCustomer();
+        if (customer == null) {
+            return ImmutableList.of();
+        }
+
+        final List<AddressInfo> addresses = new LinkedList<>();
+
+        if (customer.getDestination() != null) {
+            addresses.add(new AddressInfo(customer.getDestination()));
+        }
+
+        if (customer.getTaxRegistrations() != null && customer.getTaxRegistrations().size() > 0) {
+            addresses.addAll(customer.getTaxRegistrations()
+                                     .stream()
```

Comment:
From the sample response I had, I see both customer destination and tax-registrations are empty/nil in Vertex response? Is my understanding correct. If yes, we may have to use address from the request.

Reply:
go test -run  TestKillbillSuite/TestInvoiceTaxes_Colorado

produces this addresses:
![image](https://github.com/killbill/killbill-vertex-plugin/assets/93208596/c9bee1d2-9df0-4574-9fc4-5825bba8434b)

Reply:
I don't think we need to rely on tax registrations since we have Customer-> Destination in response.